_plugins/convert_md.rb

```diff
@@ -0,0 +1,9 @@
+# Convert the links that point to a 'README.md' to 'index.html'
+Jekyll::Hooks.register :pages, :post_render do |doc|
+  doc.output = doc.output.gsub(/(href="\.?\.\/[^.]*)README\.md?(#(.*))?"/, '\1index.html\2"')
```

Comment:
Does your regex only match `./` and `../` links? Does Jekyll ever output relative links to subfolders like `blah/`?

Reply:
yes, mostly to catch the `./` or `../` relative links. 

For the second question I don't understand what you mean. Could you provide an example please ?